### PR TITLE
feat: Message Component Interactions

### PR DESCRIPTION
**Pull request by DjDeveloperr**
This PR starts work on the Message Component Interactions (like UI elements in Messages).

### CODE REVIEW

**Review by minseo0388 — APPROVED** [review]

**Comment by minseo0388** [comment]
approved for now but not further update i will re approve when edited

**Comment by DjDeveloperr** [comment]
ready for review as buttons landed

File: src/interactions/slashCommand.ts [hunk 1]
```diff
@@ -368,7 +368,7 @@ export class SlashCommandPermissionsManager {
     )
     return data.map(transformSlashCommandPermissionsPayload)
   }
-
+
```
Comment (Helloyunho) [inline src/interactions/slashCommand.ts:371]: I think you added a bit of unnecessary bytes

Comment (DjDeveloperr) [reply src/interactions/slashCommand.ts:371]: :facepalm:

File: src/managers/_util.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  MessageComponentPayload,
+  MessageComponentType
+} from '../types/messageComponents.ts'
+
+export function transformComponent(
+  d: MessageComponentData[]
+): MessageComponentPayload[] {
+  return d.map((e: any) => {
+    e = { ...e }
+    if (e.customID !== undefined) {
```
Comment (Helloyunho) [inline src/managers/_util.ts:13]: Maybe you can use `toCamelCase()` for this

**Review by Helloyunho — APPROVED** [review]
LGTM!

**Pull request merged**